app/prepare_proposal.go

```diff
@@ -32,7 +32,34 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 		ante.DefaultSigVerificationGasConsumer,
 		app.IBCKeeper,
 	)
-	txs := FilterTxs(sdkCtx, handler, app.txConfig, req.BlockData.Txs)
+
+	var txs [][]byte
+	// This if statement verifies whether the preparation of the proposal
+	// pertains to the first block. If it does, the block is constructed using
+	// an empty set of transactions. However, even without this validation,
```

Comment:
Just double-checking my understanding: the genesis file can still contain Txs, right? They simply won't show up as they materialize in the state (and not in the block data) before PrepareProposal is even called. Is my understanding correct?

Reply:
yeah, genesis transaction are executed by the [genutil module](https://github.com/celestiaorg/cosmos-sdk/blob/c525ad72ab1449d31115815f5832a68bc79d4ffd/x/genutil/gentx.go#L92). Here we're just being overly paranoid that no one accidently attempts to propose a block with transactions in it since they'll panic.